### PR TITLE
Allow players to pay other players coins [FS-7]

### PR DESCRIPTION
**Pull request by VideoGameSmash12**
Should be good enough, but feedback would be lovely.

### CODE REVIEW

**Comment by VideoGameSmash12** [comment]
Disregard the build failure. It seems that the dependencies are on the fritz again.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java [hunk 1]
```diff
@@ -9,8 +9,11 @@
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
 
+import java.util.ArrayList;
+import java.util.List;
+
 @CommandPermissions(level = Rank.OP, source = SourceType.BOTH)
-@CommandParameters(description = "Shows the amount of coins you have or another player has", usage = "/<command> [playername]")
+@CommandParameters(description = "Shows the amount of coins you or another player has. ", usage = "/<command> [player] | pay <player> <amount>")
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java:16]: Could you update the description to be more accurate to reflect the fact it won't just show coin values.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java [hunk 2]
```diff
@@ -21,34 +24,119 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
...
-        else
-        {
-            if (senderIsConsole)
+
+            // Mode for paying another player coins
+            case 3:
             {
-                msg(prefix + ChatColor.RED + "You are not a player, use /coins <playername>");
-                return true;
+                if (args[0].equalsIgnoreCase("pay"))
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java:68]: Do we catch the else anywhere to basically throw a not found command / similar?

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java [hunk 3]
```diff
@@ -21,34 +24,119 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
...
+                    // Prevents players from performing transactions they can't afford to do.
+                    if (senderData.getCoins() < coinsToTransfer)
+                    {
+                        msg("You don't have enough coins to perform this transaction.", ChatColor.RED);
+                        return true;
+                    }
+
+                    if (target == null)
+                    {
+                        msg(PLAYER_NOT_FOUND);
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java:98]: Do we not also need to return true out of this statement?

**Review by Wild1145 — CHANGES_REQUESTED** [review]

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java (hunk 3 above)
Comment (VideoGameSmash12) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java:98]: Already a thing at line 117

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java (hunk 2 above)
Comment (VideoGameSmash12) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java:68]: Yes; it returns false at line 121

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java (hunk 1 above)
Comment (VideoGameSmash12) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_coins.java:16]: good catch, though i'm not exactly sure how i'd word it

**Comment by VideoGameSmash12** [comment]
pinged because of your original review

**Review by Focusvity — APPROVED** [review]
LGTM

**Pull request merged**